### ISSUE TITLE
1-D convolution of tensorflow do not support float64 exactly.

### ISSUE THREAD

**Issue by loewy43** [tensorflow/tensorflow#19175]
In the description of tf.nn.conv1d, it said "value: A 3D Tensor. Must be of type float32 or float64". But the float64 isn't supported correctly. Some one could try the code below:

import numpy as np
import tensorflow as tf
a = np.random.randn(12, 228, 1)
w = np.random.randn(2, 1, 18)
sess = tf.Session()
c = sess.run(tf.nn.conv1d(tf.cast(a, tf.float64), tf.cast(w, tf.float64), stride=1, padding='SAME'))
sess.close()

I have test it in tensorflow 1.4 and 1.7,  they raised the error : 
Value passed to parameter 'input' has DataType float64 not in list of allowed values: float16, float32

**Comment by yongtang** [comment]
@loewy43 The float64 support for conv2d has been added e3468b5. The conv1d implementation invokes conv2d, so float64 support for conv1d is already available as well. (The support starts with 1.8.0).

I have added a test case in #19179 to cover the conv1d.
